src/unit-default-test-run-workspace/unit-default-test-run.service.ts

```diff
@@ -66,10 +67,13 @@ export class UnitDefaultTestRunWorkspaceService {
     userId: string,
     isImport: boolean = false,
     historicalRecordId?: string,
+    trx?: EntityManager,
   ): Promise<UnitDefaultTestRunRecordDTO> {
     const timestamp = currentDateTime();
 
-    let entity = this.repository.create({
+    const repo = trx ? trx.getRepository(UnitDefaultTestRun) : this.repository;
```

Comment:
In calling `trx.getRepository(UnitDefaultTestRun)`, **you are targeting the official repository when creating a new entity** (since `this.repository` points to the "workspace" version).

This is another reason why I recommend using the `withTransaction` utility, instead. I only caught this at the end of this review, so this may be an issue elsewhere, too. I would recommend going back through all other locations that use this pattern and either: A) switch to the custom `withTransaction` utility or B) ensure the repositories align.